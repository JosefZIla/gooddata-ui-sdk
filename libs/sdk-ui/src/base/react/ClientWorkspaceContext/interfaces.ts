// (C) 2019-2022 GoodData Corporation

/**
 * Resolved LCM identifiers of the workspace.
 *
 * @alpha
 */
export interface IClientWorkspaceIdentifiers {
    /**
     * Data product identifier.
     */
    dataProduct?: string;

    /**
     * Client identifier.
     */
    client?: string;

    /**
     * Segment identifier.
     */
    segment?: string;

    /**
     * Workspace identifier.
     */
    workspace?: string;
}

/**
 * @internal
 */
export interface IClientWorkspaceStatus {
    isInitialized: boolean;
}
