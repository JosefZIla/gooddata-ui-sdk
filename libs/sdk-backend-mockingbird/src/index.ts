// (C) 2019-2021 GoodData Corporation
/**
 * @packageDocumentation
 */
import { dummyBackend, dummyBackendEmptyData, dummyDataView } from "@gooddata/sdk-backend-base";
export { dummyBackend, dummyBackendEmptyData, dummyDataView };

export { recordedBackend } from "./recordedBackend";

export {
    RecordingIndex,
    ExecutionRecording,
    InsightRecording,
    DisplayFormRecording,
    ScenarioRecording,
    RecordedBackendConfig,
    CatalogRecording,
    VisClassesRecording,
    DashboardRecording,
    RecordedRefType,
    SecuritySettingsUrlValidator,
    SecuritySettingsOrganizationScope,
} from "./recordedBackend/types";

export {
    recordedDataView,
    recordedDataViews,
    recordedInsight,
    DataViewAll,
    dataViewWindow,
    DataViewFirstPage,
    NamedDataView,
} from "./recordedBackend/execution";

export { CompositeBackendPart, compositeBackend } from "./compositeBackend/index";

export {
    LegacyExecutionRecording,
    legacyRecordedBackend,
    legacyRecordedDataView,
    LegacyRecordingIndex,
    LegacyWorkspaceRecordings,
} from "./legacyRecordedBackend";
