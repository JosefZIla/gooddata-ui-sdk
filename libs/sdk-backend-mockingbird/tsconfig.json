{
    "compilerOptions": {
        "esModuleInterop": true,
        "target": "es5",
        "module": "commonjs",
        "moduleResolution": "node",
        "lib": ["es6", "dom", "ES2016.Array.Include"],
        "importHelpers": true,
        "declaration": true,
        "declarationMap": true,
        "sourceMap": true,
        "outDir": "./dist",
        "strict": true,
        "noImplicitAny": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "noImplicitReturns": true,
        "baseUrl": "..",
        "paths": {
            "@gooddata/sdk-model": ["sdk-model/src"],
            "@gooddata/sdk-backend-base": ["sdk-backend-base/src"],
            "@gooddata/sdk-backend-spi": ["sdk-backend-spi/src"]
        }
    },
    "exclude": ["node_modules", "dist"]
}
