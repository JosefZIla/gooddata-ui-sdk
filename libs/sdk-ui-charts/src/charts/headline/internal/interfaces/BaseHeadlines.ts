// (C) 2023 GoodData Corporation
import { HeadlineElementType } from "@gooddata/sdk-ui";
import { IHeadlineDataItem } from "./Headlines.js";
import { ComponentType, RefObject } from "react";

export type BaseHeadlineDataItemComponentType = ComponentType<
    IBaseHeadlineDataItemProps & IWithDrillableItemProps & IWithTitleProps
>;

export interface IBaseHeadlineItem {
    data: IHeadlineDataItem;
    baseHeadlineDataItemComponent: BaseHeadlineDataItemComponentType;
    elementType?: HeadlineElementType;
}

export interface IBaseHeadlineData {
    primaryItem: IBaseHeadlineItem;
    secondaryItem?: IBaseHeadlineItem;
    tertiaryItem?: IBaseHeadlineItem;
}

export interface IBaseHeadlineDataItemProps {
    dataItem: IHeadlineDataItem;
}

export interface IWithDrillableItemProps {
    dataItem: IHeadlineDataItem;
    elementType?: HeadlineElementType;
}

export interface IWithTitleProps {
    dataItem: IHeadlineDataItem;
    shouldHideTitle?: boolean;
    titleRef?: RefObject<HTMLDivElement>;
}
