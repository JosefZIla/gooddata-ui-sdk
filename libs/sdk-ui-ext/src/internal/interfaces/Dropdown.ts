// (C) 2019-2021 GoodData Corporation
import { SingleSelectListItemType } from "@gooddata/sdk-ui-kit";

import translations from "../translations/en-US.json";

export interface IDropdownItem {
    title?: string;
    value?: string | number | boolean;
    type?: SingleSelectListItemType;
    icon?: string;
    info?: string;
}

type TranslationKeys = keyof typeof translations;

/**
 * This type narrows down the title to be a key from the translations bundle. Otherwise it provides
 * same properties as the normal dropdown.
 */
export interface ITranslatedDropdownItem extends IDropdownItem {
    title?: TranslationKeys;
}
