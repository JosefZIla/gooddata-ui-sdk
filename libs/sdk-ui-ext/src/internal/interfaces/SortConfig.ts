// (C) 2019-2022 GoodData Corporation

import { Identifier, IAttributeLocatorItem, newMeasureSort } from "@gooddata/sdk-model";
import { MeasureSortSuggestion } from "@gooddata/sdk-ui-kit";

export function newMeasureSortSuggestion(
    identifier: Identifier,
    attributeLocators: IAttributeLocatorItem[] = [],
): MeasureSortSuggestion {
    const {
        measureSortItem: { locators },
    } = newMeasureSort(identifier, "asc", attributeLocators);
    return {
        type: "measureSort",
        locators,
    };
}
