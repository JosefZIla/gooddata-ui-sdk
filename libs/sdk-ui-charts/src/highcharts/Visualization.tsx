// (C) 2007-2018 GoodData Corporation
import { IDataView } from "@gooddata/sdk-backend-spi";
import invariant from "ts-invariant";
import React from "react";
import { IDrillableItem, OnFiredDrillEvent, IHeaderPredicate } from "@gooddata/sdk-ui";
import { ChartTransformation, renderHighCharts } from "./ChartTransformation";
import Highcharts from "./lib";
import { IChartConfig } from "../interfaces";

import { isChartSupported, stringifyChartTypes } from "./chartTypes/_util/common";
import isEqual from "lodash/isEqual";
import isFunction from "lodash/isFunction";
import noop from "lodash/noop";
import omitBy from "lodash/omitBy";

/**
 * @internal
 */
export interface IVisualizationProps {
    height: number;
    width: number;
    config: IChartConfig;
    numericSymbols?: string[];

    dataView: IDataView;
    drillableItems: Array<IDrillableItem | IHeaderPredicate>;
    locale?: string;

    onDrill?: OnFiredDrillEvent;
    afterRender?: () => void;
    onDataTooLarge(): void;
    onNegativeValues(): void;
    onLegendReady(): void;
    pushData?(data: any): void;
}
// TODO: REMOVE VISUALIZATION COMPONENT
/**
 * @internal
 */
export class Visualization extends React.Component<IVisualizationProps> {
    public static defaultProps = {
        locale: "en-US",
        numericSymbols: [] as string[],
        onDrill: (): boolean => true,
        afterRender: noop,
    };

    constructor(props: IVisualizationProps) {
        super(props);

        this.setNumericSymbols(this.props);
    }

    public UNSAFE_componentWillReceiveProps(nextProps: IVisualizationProps): void {
        this.setNumericSymbols(nextProps);
    }

    public shouldComponentUpdate(nextProps: IVisualizationProps): boolean {
        return !isEqual(omitBy(this.props, isFunction), omitBy(nextProps, isFunction));
    }

    public setNumericSymbols(props: IVisualizationProps): void {
        const { numericSymbols } = props;

        if (numericSymbols && numericSymbols.length) {
            Highcharts.setOptions({
                lang: {
                    numericSymbols,
                },
            });
        }
    }

    public render(): React.ReactNode {
        const visType = this.props.config.type;

        if (isChartSupported(visType)) {
            const {
                height,
                width,
                config,
                dataView,
                drillableItems,
                onDrill,
                afterRender,
                onDataTooLarge,
                onNegativeValues,
                onLegendReady,
                locale,
                pushData,
            } = this.props;

            return (
                <ChartTransformation
                    height={height}
                    width={width}
                    config={config}
                    drillableItems={drillableItems}
                    locale={locale}
                    dataView={dataView}
                    afterRender={afterRender}
                    onDrill={onDrill}
                    onDataTooLarge={onDataTooLarge}
                    onNegativeValues={onNegativeValues}
                    onLegendReady={onLegendReady}
                    pushData={pushData}
                    renderer={renderHighCharts}
                />
            );
        }

        invariant(
            false,
            `Unknown visualization type: ${visType}. Supported visualization types: ${stringifyChartTypes()}`,
        );

        return null;
    }
}
