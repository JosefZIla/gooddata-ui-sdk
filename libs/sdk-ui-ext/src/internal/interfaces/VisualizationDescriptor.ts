// (C) 2021 GoodData Corporation
import { ISettings } from "@gooddata/sdk-backend-spi";
import { IInsightDefinition } from "@gooddata/sdk-model";
import { IFluidLayoutDescriptor } from "./LayoutDescriptor";
import { IVisConstruct, IVisualization } from "./Visualization";

/**
 * Factories that create a new instance of pluggable visualization.
 *
 * @alpha
 */
export type PluggableVisualizationFactory = (params: IVisConstruct) => IVisualization;

/**
 * Info about min, max and default in layout units.
 * If no value is provided for min/max/default layout will use its generic defaults for it.
 *
 * @alpha
 */
interface ISizeInfo {
    default?: number;
    min?: number;
    max?: number;
}

/**
 * Info about visualization's min, max and default for width and height.
 *
 * @alpha
 */
export interface IVisualizationSizeInfo {
    width: ISizeInfo;
    height: ISizeInfo;
}

/**
 * @alpha
 * Visualization descriptor
 * Provides access to the visualization's factory and additional details about visualization
 */
export interface IVisualizationDescriptor {
    /**
     * Gets pluggable visualization factory.
     *
     * @alpha
     */
    getFactory(): PluggableVisualizationFactory;

    /**
     * Gets info about size of visualization based on its type, bucket content, layout context and if custom height is enabled.
     * Provided info is used by layout and it defines what will be the default size during render of insight and what will be the limits during resizing of insight.
     *
     * @param insight Current insight definition
     * @param layoutDescriptor layout creating context for visualization
     * @param settings to know if custom heights are allowed
     * @alpha
     */
    getSizeInfo(
        insight: IInsightDefinition,
        layoutDescriptor: IFluidLayoutDescriptor,
        settings: ISettings,
    ): IVisualizationSizeInfo;
}
