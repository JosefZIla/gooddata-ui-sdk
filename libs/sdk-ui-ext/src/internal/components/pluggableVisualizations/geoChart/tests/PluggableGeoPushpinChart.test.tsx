// (C) 2019-2022 GoodData Corporation
import noop from "lodash/noop";
import * as referencePointMocks from "../../../../tests/mocks/referencePointMocks";
import { PluggableGeoPushpinChart } from "../PluggableGeoPushpinChart";
import { IExtendedReferencePoint, IVisConstruct } from "../../../../interfaces/Visualization";
import { dummyBackend } from "@gooddata/sdk-backend-mockingbird";
import { getLastRenderEl } from "../../tests/testHelpers";
import * as testMocks from "../../../../tests/mocks/testMocks";

describe("PluggableGeoPushpinChart", () => {
    const mockElement = document.createElement("div");
    const mockConfigElement = document.createElement("div");
    const mockRenderFun = jest.fn();
    const executionFactory = dummyBackend().workspace("PROJECTID").execution();
    const defaultProps: IVisConstruct = {
        projectId: "PROJECTID",
        element: () => mockElement,
        configPanelElement: () => mockConfigElement,
        callbacks: {
            afterRender: noop,
            pushData: noop,
        },
        backend: dummyBackend(),
        visualizationProperties: {},
        renderFun: mockRenderFun,
    };

    function createComponent(props: IVisConstruct = defaultProps) {
        return new PluggableGeoPushpinChart(props);
    }

    afterEach(() => {
        mockRenderFun.mockReset();
    });

    it("should create geo pushpin visualization", () => {
        const visualization = createComponent();

        expect(visualization).toBeTruthy();
    });

    describe("getExtendedReferencePoint", () => {
        const geoPushpin = createComponent();
        const sourceReferencePoint = referencePointMocks.simpleGeoPushpinReferencePoint;
        const extendedReferencePointPromise: Promise<IExtendedReferencePoint> =
            geoPushpin.getExtendedReferencePoint(sourceReferencePoint);

        it("should return a new reference point with geoPushpin adapted buckets", () => {
            return extendedReferencePointPromise.then((extendedReferencePoint) => {
                expect(extendedReferencePoint.buckets).toEqual(sourceReferencePoint.buckets);
            });
        });

        it("should return a new reference point with geoPushpin UI config", () => {
            return extendedReferencePointPromise.then((extendedReferencePoint) => {
                expect(extendedReferencePoint.uiConfig).toMatchSnapshot();
            });
        });

        it("should transform view by attribute to location attribute", async () => {
            const { oneMetricAndGeoCategoryAndStackReferencePoint } = referencePointMocks;

            const newExtendedReferencePoint = await geoPushpin.getExtendedReferencePoint(
                oneMetricAndGeoCategoryAndStackReferencePoint,
            );

            expect(newExtendedReferencePoint.buckets).toEqual([
                {
                    localIdentifier: "location",
                    items: [
                        {
                            localIdentifier: "a1",
                            type: "attribute",
                            aggregation: null,
                            attribute: "attr.owner.country",
                            locationDisplayFormRef: { uri: "/geo/attribute/displayform/uri/1" },
                            dfRef: { uri: "/geo/attribute/displayform/uri/2" },
                        },
                    ],
                },
                {
                    localIdentifier: "size",
                    items: [
                        {
                            localIdentifier: "m1",
                            type: "metric",
                            aggregation: null,
                            attribute: "aazb6kroa3iC",
                            showInPercent: null,
                            showOnSecondaryAxis: null,
                        },
                    ],
                },
                {
                    localIdentifier: "color",
                    items: [],
                },
                {
                    localIdentifier: "segment",
                    items: [
                        {
                            localIdentifier: "a2",
                            type: "attribute",
                            aggregation: null,
                            attribute: "attr.stage.iswon",
                            dfRef: {
                                uri: "a2/df",
                            },
                        },
                    ],
                },
            ]);
        });

        it("should transform geo attribute with attribute", async () => {
            const { viewByWithDateAndGeoAttributeReferencePoint } = referencePointMocks;

            const newExtendedReferencePoint = await geoPushpin.getExtendedReferencePoint(
                viewByWithDateAndGeoAttributeReferencePoint,
            );

            expect(newExtendedReferencePoint.buckets).toEqual([
                {
                    localIdentifier: "location",
                    items: [
                        {
                            localIdentifier: "a1",
                            type: "attribute",
                            aggregation: null,
                            attribute: "attr.owner.country",
                            locationDisplayFormRef: { uri: "/geo/attribute/displayform/uri/1" },
                            dfRef: { uri: "/geo/attribute/displayform/uri/2" },
                        },
                    ],
                },
                {
                    localIdentifier: "size",
                    items: [
                        {
                            localIdentifier: "m1",
                            type: "metric",
                            aggregation: null,
                            attribute: "aazb6kroa3iC",
                            showInPercent: null,
                            showOnSecondaryAxis: null,
                        },
                    ],
                },
                {
                    localIdentifier: "color",
                    items: [],
                },
                {
                    localIdentifier: "segment",
                    items: [],
                },
            ]);
        });

        it("should not transform same location attribute on location and segment bucket", async () => {
            const { viewByWithNonGeoAndGeoAttributeReferencePoint } = referencePointMocks;

            const newExtendedReferencePoint = await geoPushpin.getExtendedReferencePoint(
                viewByWithNonGeoAndGeoAttributeReferencePoint,
            );

            expect(newExtendedReferencePoint.buckets).toEqual([
                {
                    localIdentifier: "location",
                    items: [
                        {
                            localIdentifier: "a1",
                            type: "attribute",
                            aggregation: null,
                            attribute: "attr.owner.country",
                            locationDisplayFormRef: { uri: "/geo/attribute/displayform/uri/1" },
                            dfRef: { uri: "/geo/attribute/displayform/uri/2" },
                        },
                    ],
                },
                {
                    localIdentifier: "size",
                    items: [
                        {
                            localIdentifier: "m1",
                            type: "metric",
                            aggregation: null,
                            attribute: "aazb6kroa3iC",
                            showInPercent: null,
                            showOnSecondaryAxis: null,
                        },
                    ],
                },
                {
                    localIdentifier: "color",
                    items: [],
                },
                {
                    localIdentifier: "segment",
                    items: [
                        {
                            aggregation: null,
                            attribute: "attr.owner.department",
                            dfRef: {
                                uri: "a1/df",
                            },
                            localIdentifier: "a1",
                            type: "attribute",
                        },
                    ],
                },
            ]);
        });

        it("should reset showInPercent and showOnSecondaryAxis for size and color measures", async () => {
            const newExtendedReferencePoint = await geoPushpin.getExtendedReferencePoint(
                referencePointMocks.twoMeasuresWithShowInPercentOnSecondaryAxisReferencePoint,
            );
            expect(newExtendedReferencePoint.buckets).toEqual([
                {
                    localIdentifier: "location",
                    items: [
                        {
                            aggregation: null,
                            attribute: "attr.owner.country",
                            dfRef: { uri: "/geo/attribute/displayform/uri/2" },
                            localIdentifier: "a1",
                            locationDisplayFormRef: { uri: "/geo/attribute/displayform/uri/1" },
                            type: "attribute",
                        },
                    ],
                },
                {
                    localIdentifier: "size",
                    items: [
                        {
                            localIdentifier: "m3",
                            type: "metric",
                            aggregation: null,
                            attribute: "dt.opportunitysnapshot.snapshotdate",
                            showInPercent: null,
                            showOnSecondaryAxis: null,
                        },
                    ],
                },
                {
                    localIdentifier: "color",
                    items: [
                        {
                            localIdentifier: "m4",
                            type: "metric",
                            aggregation: null,
                            attribute: "acfWntEMcom0",
                            showInPercent: null,
                            showOnSecondaryAxis: null,
                        },
                    ],
                },
                {
                    localIdentifier: "segment",
                    items: [],
                },
            ]);
        });

        it("should return a new reference point with geoPushpin supported properties list", async () => {
            const geoPushpinChart = createComponent();
            const extendedReferencePoint = await geoPushpinChart.getExtendedReferencePoint(
                referencePointMocks.simpleGeoPushpinReferencePoint,
            );
            expect(extendedReferencePoint.properties).toEqual({
                controls: {
                    points: {
                        groupNearbyPoints: false,
                    },
                    tooltipText: "/geo/attribute/displayform/uri/2",
                },
            });
        });

        it("should remove all derived measures", async () => {
            const geoPushpinChart = createComponent();
            const extendedReferencePoint = await geoPushpinChart.getExtendedReferencePoint(
                referencePointMocks.samePeriodPreviousYearRefPoint,
            );

            expect(extendedReferencePoint.buckets).toEqual([
                {
                    localIdentifier: "location",
                    items: [],
                },
                {
                    localIdentifier: "size",
                    items: [
                        {
                            localIdentifier: "m1",
                            type: "metric",
                            aggregation: null,
                            attribute: "aazb6kroa3iC",
                            showInPercent: null,
                            showOnSecondaryAxis: null,
                        },
                    ],
                },
                {
                    localIdentifier: "color",
                    items: [],
                },
                {
                    localIdentifier: "segment",
                    items: [],
                },
            ]);
        });

        it("should remove all arithmetic measures", async () => {
            const geoPushpinChart = createComponent();
            const extendedReferencePoint = await geoPushpinChart.getExtendedReferencePoint(
                referencePointMocks.firstMeasureArithmeticNoAttributeReferencePoint,
            );

            expect(extendedReferencePoint.buckets).toEqual([
                {
                    localIdentifier: "location",
                    items: [],
                },
                {
                    localIdentifier: "size",
                    items: [
                        {
                            localIdentifier: "m1",
                            type: "metric",
                            aggregation: null,
                            attribute: "aazb6kroa3iC",
                            showInPercent: null,
                            showOnSecondaryAxis: null,
                        },
                    ],
                },
                {
                    localIdentifier: "color",
                    items: [
                        {
                            localIdentifier: "m2",
                            type: "metric",
                            aggregation: null,
                            attribute: "af2Ewj9Re2vK",
                            showInPercent: null,
                            showOnSecondaryAxis: null,
                        },
                    ],
                },
                {
                    localIdentifier: "segment",
                    items: [],
                },
            ]);
        });
    });

    describe("`renderVisualization` and `renderConfigurationPanel`", () => {
        it("should mount on the element defined by the callback", () => {
            const visualization = createComponent();

            visualization.update({}, testMocks.insightWithSingleMeasure, {}, executionFactory);

            // 1st call for rendering element
            // 2nd call for rendering config panel
            expect(mockRenderFun).toHaveBeenCalledTimes(2);
            expect(getLastRenderEl(mockRenderFun, mockElement)).toBeDefined();
            expect(getLastRenderEl(mockRenderFun, mockConfigElement)).toBeDefined();
        });
    });
});
