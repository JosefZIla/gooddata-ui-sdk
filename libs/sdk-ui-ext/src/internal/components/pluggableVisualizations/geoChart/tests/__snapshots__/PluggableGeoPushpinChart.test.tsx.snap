// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluggableGeoPushpinChart getExtendedReferencePoint should return a new reference point with geoPushpin UI config 1`] = `
Object {
  "buckets": Object {
    "color": Object {
      "accepts": Array [
        "metric",
        "fact",
        "attribute",
      ],
      "allowsDuplicateItems": true,
      "allowsReordering": false,
      "allowsSwapping": true,
      "canAddItems": true,
      "enabled": true,
      "icon": "local:geoPushpin/bucket-title-color-icon.svg",
      "isShowInPercentEnabled": false,
      "isShowInPercentVisible": false,
      "itemsLimit": 1,
      "subtitle": "Color",
      "title": "Measure",
    },
    "filters": Object {
      "accepts": Array [
        "attribute",
        "date",
      ],
      "allowsReordering": false,
      "enabled": true,
      "isShowInPercentEnabled": false,
      "itemsLimit": 20,
      "itemsLimitByType": Object {
        "date": 1,
      },
    },
    "location": Object {
      "accepts": Array [
        "attribute",
        "geo_attribute",
      ],
      "allowsReordering": false,
      "allowsSwapping": true,
      "canAddItems": true,
      "enabled": true,
      "icon": "local:geoPushpin/bucket-title-location-icon.svg",
      "isShowInPercentEnabled": false,
      "itemsLimit": 1,
      "itemsLimitByType": Object {
        "date": 1,
      },
      "title": "Location",
    },
    "segment": Object {
      "accepts": Array [
        "attribute",
      ],
      "allowsReordering": false,
      "allowsSwapping": true,
      "canAddItems": true,
      "enabled": true,
      "icon": "local:geoPushpin/bucket-title-segment-icon.svg",
      "isShowInPercentEnabled": false,
      "itemsLimit": 1,
      "itemsLimitByType": Object {
        "date": 1,
      },
      "title": "Segment by",
    },
    "size": Object {
      "accepts": Array [
        "metric",
        "fact",
        "attribute",
      ],
      "allowsDuplicateItems": true,
      "allowsReordering": false,
      "allowsSwapping": true,
      "canAddItems": true,
      "enabled": true,
      "icon": "local:geoPushpin/bucket-title-size-icon.svg",
      "isShowInPercentEnabled": false,
      "isShowInPercentVisible": false,
      "itemsLimit": 1,
      "subtitle": "Size",
      "title": "Measure",
    },
  },
  "exportConfig": Object {
    "supported": true,
  },
  "noMetricAccepted": Object {
    "supported": true,
  },
  "openAsReport": Object {
    "supported": false,
  },
  "recommendations": Object {},
  "supportedLocationIcon": Object {
    "supported": true,
  },
  "supportedOverTimeComparisonTypes": Array [],
}
`;
